internal/web/server.go

```diff
@@ -100,6 +104,9 @@ type server struct {
 	packageController     cache.Controller
 	packageInfoStore      cache.Store
 	packageInfoController cache.Controller
+	namespaceLister       *v1.NamespaceLister
+	configMapLister       *v1.ConfigMapLister
+	secretLister          *v1.SecretLister
```

Comment:
I'm not sure about best practices here, but since this feature is based on shared informers, wouldn't it be easier to just store the factory in the server object and then create listers on demand?

Reply:
The problem with this approach is that the factory has to be started at some point, and inside the `Start()` method it needs to know the registered informers. When creating a lister with this factory after it has been started, it never returns any results because the informer is never started. 
We could initialize the three relevant listers inside `initWhenBootstrapped` before calling `Start()`, like this:
```
factory := informers.NewSharedInformerFactory(server.k8sClient, 0)
c := make(chan struct{})
factory.Core().V1().ConfigMaps().Lister()
factory.Start(c)
server.informerFactory = &factory
```
This way it would work by getting the listers "on demand" (which they wouldn't really be, because they are stored in the factory anyway), but I don't like it because if you need another lister at some other point in the future, you need to be aware of initializing it here first. Plus it doesn't seem right to request a lister here and then do nothing with it? But maybe I'm missing something.

Reply:
Ohh, I didn't catch that! I thould it would actually start the informers on demand. 

In that case, I think the current approach is fine.